cohttp-eio/src/io.ml

```diff
@@ -0,0 +1,40 @@
+module IO = struct
+  type 'a t = 'a
+
+  let ( >>= ) v f = f v
+  let return v = v
+
+  type ic = Eio.Buf_read.t
+  type oc = Eio.Buf_write.t
```

Comment:
Maybe it would also make sense to have:

```
type ic = Eio.Flow.sink
type oc = Eio.Flow.source
```

This would maybe allow directly streaming a file (see https://github.com/mirage/ocaml-cohttp/issues/970) and ping @TheLortex.

Reply:
It's been some time since I wrote this, but I remember I started with these and it didn't work out. AFAIR multiple functions from `IO` are called successively, and one creating then discarding eg. a `Buf_read.t` over a `Flow.source` each time would lose the buffered data. It seems to me the correct data type to persist the read state is the buffer, which includes the stateful buffered data and cursor position.